http/src/v0.rs

```diff
@@ -1,3 +1,132 @@
-pub mod version;
-// pub mod id;
+use ipfs::{Ipfs, IpfsTypes};
+use std::convert::Infallible;
+
+pub mod id;
 pub mod swarm;
+pub mod version;
+
+pub mod support;
+pub use support::recover_as_message_response;
+pub(crate) use support::{with_ipfs, InvalidPeerId, NotImplemented, StringError};
+
+pub fn routes<T>(
+    ipfs: &Ipfs<T>,
+    shutdown_tx: tokio::sync::mpsc::Sender<()>,
+) -> impl warp::Filter<Extract = impl warp::Reply, Error = Infallible> + Clone
+where
+    T: IpfsTypes,
+{
+    use warp::{query, Filter};
+    // /api/v0/shutdown
+    let shutdown = warp::post()
+        .and(warp::path!("shutdown"))
+        .and(warp::any().map(move || shutdown_tx.clone()))
+        .and_then(handle_shutdown);
+
+    let mount = warp::path("api").and(warp::path("v0"));
+
+    let api = mount.and(
+        shutdown
+            .or(id::identity(ipfs))
```

Comment:
because of the ... special casing of the arg handling I thought this might be the best way, `identity` is now fully built filter inside the module. Not so great for reusability but at least the modifiers are not in conflict (compiler error). 

Obviously we need to figure out a way to have those "remote" types be `serde::Deserializable` but also didn't want to start PR:ing every multiformat project until we know better.